runtime/gc_vlhgc/IncrementalGenerationalGC.cpp

```diff
@@ -1203,8 +1212,15 @@ MM_IncrementalGenerationalGC::partialGarbageCollect(MM_EnvironmentVLHGC *env, MM
 	if (_schedulingDelegate.isGlobalSweepRequired()) {
 		Assert_MM_true(NULL == env->_cycleState->_externalCycleState);
 
+		PORT_ACCESS_FROM_ENVIRONMENT(env);
+
+		U_64 globalSweepStartTime = j9time_hires_clock();
+
 		_reclaimDelegate.runGlobalSweepBeforePGC(env, allocDescription, env->_cycleState->_activeSubSpace, env->_cycleState->_gcCode);
 
+		/** Time indicated here is ~10% higher than that of corresponding sweeps timems in vgc logs - which is okay. It is factored into GMP overhead */
+		U_64 globalSweepTimeUs = j9time_hires_delta(globalSweepStartTime, j9time_hires_clock(), J9PORT_TIME_DELTA_IN_MICROSECONDS);
```

Comment:
could we reuse start/end TS recorded by MM_ReclaimDelegate::doSweep() ?

Reply:
I don't see sweep duration being reported by trace (neither here, not later when you use it for GC overhead calculation). If so, please  expand Trc_MM_SweepEnd to report it (you'll have to obsolete the existing one and create V2)

Reply:
To minimize changes in OMR sweep code you could keep both version, and just make Balanced used the new one.